### PR TITLE
 HHH-10418 Support entity/collection sharing same region

### PR DESCRIPTION
**Pull request by galderz**
https://hibernate.atlassian.net/browse/HHH-10418

### CODE REVIEW

File: hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/EntitiesAndCollectionsInSameRegionTest.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+				AnEntity.class,
+				AnotherEntity.class
+		};
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	protected void addSettings(Map settings) {
+		super.addSettings( settings );
+		settings.put( Environment.CACHE_KEYS_FACTORY, DefaultCacheKeysFactory.SHORT_NAME );
```
Comment (rvansa) [inline hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/EntitiesAndCollectionsInSameRegionTest.java]: This should be default in master, isn't it?

Comment (galderz) [reply hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/EntitiesAndCollectionsInSameRegionTest.java]: The test uses a slightly different value:

    settings.put( Environment.CACHE_KEYS_FACTORY, DefaultCacheKeysFactory.SHORT_NAME )

I'm testing to see if the test demands this.

Comment (galderz) [reply hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/EntitiesAndCollectionsInSameRegionTest.java]: If removed it fails with:

    org.hibernate.WrongClassException: Object [id=1] was not of the specified subclass...

**Comment by gbadner** [comment]
Merged in 5.2 branch.

**Pull request closed without merging**